### PR TITLE
Add Sanity tests products BO

### PR DESCRIPTION
**Pull request by boubkerbribri**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Testing products deleting methods 
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | `TEST_PATH="sanity/productsBO/*" URL_FO=SHOP_URL npm run specific-test`

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/15422)

### CODE REVIEW

File: tests/puppeteer/campaigns/sanity/productsBO/04_deleteProduct.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  test('should reset all filters', async () => {
+    if (await productsPage.elementVisible(productsPage.filterResetButton, 2000)) await productsPage.resetFilter();
+    await productsPage.resetFilterCategory();
+    const numberOfProducts = await productsPage.getNumberOfProductsFromList();
+    await expect(numberOfProducts).to.be.above(0);
+  });
+  test('should create Product', async () => {
+    await productsPage.goToAddProductPage();
+    const createProductMessage = await addProductPage.createEditProduct(productData);
+    await expect(createProductMessage).to.equal('Settings updated.');
```
Comment (SimonGrn) [inline tests/puppeteer/campaigns/sanity/productsBO/04_deleteProduct.js]: Why is the string written litterally and two tests below it's taken from an object (`productsPage.productDeletedSuccessfulMessage`) ?

File: tests/puppeteer/pages/BO/products.js [hunk 2]
```diff
@@ -123,4 +134,55 @@ module.exports = class Product extends BOBasePage {
...
+   * Delete product with dropdown Menu
+   * @param productData
+   * @return {Promise<textContent>}
+   */
+  async deleteProduct(productData) {
+    // Filter By reference first
+    await this.filterProducts('reference', productData.reference);
+    // Then delete first product and only product shown
+    await Promise.all([
+      this.page.waitForSelector(`${this.dropdownToggleButton}[aria-expanded='true']`.replace('%ROW', '1')),
```
Comment (SimonGrn) [inline tests/puppeteer/pages/BO/products.js:148]: I don't understand the `.replace` part ?

File: tests/puppeteer/pages/BO/products.js [hunk 3]
```diff
@@ -123,4 +134,55 @@ module.exports = class Product extends BOBasePage {
...
+      this.page.click(this.dropdownToggleButton.replace('%ROW', '1')),
+    ]);
+    await Promise.all([
+      this.page.waitForSelector(this.catalogDeletionModalDialog, {visible: true}),
+      this.page.click(this.dropdownMenuDeleteLink.replace('%ROW', '1')),
+    ]);
+    await Promise.all([
+      this.page.waitForNavigation({waitUntil: 'networkidle0'}),
+      this.page.waitForSelector(this.alertSuccessBlocParagraph, {visible: true}),
```
Comment (SimonGrn) [inline tests/puppeteer/pages/BO/products.js]:
```suggestion
      this.page.waitForSelector(this.alertSuccessBlockParagraph, {visible: true}),
```

**Review by SimonGrn — COMMENTED** [review]
Thanks @boubkerbribri

File: tests/puppeteer/pages/BO/products.js (hunk 2 above)
Comment (boubkerbribri) [reply tests/puppeteer/pages/BO/products.js:148]: `this.dropdownToggleButton` exists in all rows, `replace` will define which row to use.

File: tests/puppeteer/campaigns/sanity/productsBO/04_deleteProduct.js (hunk 1 above)
Comment (boubkerbribri) [reply tests/puppeteer/campaigns/sanity/productsBO/04_deleteProduct.js]: done, thanks

File: tests/puppeteer/pages/BO/products.js (hunk 3 above)
Comment (boubkerbribri) [reply tests/puppeteer/pages/BO/products.js]: Done, thanks

**Review by matthieu-rolland — APPROVED** [review]
lgtm

**Pull request merged**